ARCs/arc-0003.md

```diff
@@ -29,18 +29,30 @@ The ASA parameters should follow the following conventions:
 
 * *Unit Name* (`un`): no restriction but **SHOULD** be related to the name in the JSON Metadata file
 * *Asset Name* (`an`): **MUST** be:
-    * either exactly `arc3` (without any space)
-    * or `<name>@arc3`, where `<name>` **SHOULD** be closely related to the name in the JSON Metadata file:
+    * (**NOT RECOMMENDED**) either exactly `arc3` (without any space)
+    * (**NOT RECOMMENDED**) or `<name>@arc3`, where `<name>` **SHOULD** be closely related to the name in the JSON Metadata file:
         * If the resulting asset name can fit the *Asset Name* field, then `<name>` **SHOULD** be equal to the name in the JSON Metadata file.
         * If the resulting asset name cannot fit the *Asset Name* field, then `<name>` **SHOULD** be a reasonable shorten version of the name in the JSON Metadata file.
-* *Asset URL* (`au`): a URI pointing to a JSON Metadata file. This URI as well as any URI in the JSON Metadata file:
-    * **SHOULD** be persistent and allow to download the JSON Metadata file forever.
-    * **MAY** contain the string `{id}`. If `{id}` exists in the URI, clients **MUST** replace this with the asset ID in decimal form. The rules below applies after such a replacement.
-    * **MUST** follow [RFC-3986](https://www.ietf.org/rfc/rfc3986.txt) and **MUST NOT** contain any whitespace character
-    * **SHOULD** use one of the following URI schemes (for compatibility and security): *https* and *ipfs*:
-        * When the file is stored on IPFS, the `ipfs://...` URI **SHOULD** be used. IPFS Gateway URI (such as `https://ipfs.io/ipfs/...`) **SHOULD NOT** be used.
-    * **SHOULD NOT** use the following URI scheme: *http* (due to security concerns).
-    * **MAY** be a relative URI when inside the JSON Metadata file. In that case, the relative URI is relative to the Asset URL. The Asset URL **SHALL NOT** be relative. Relative URI **MUST** not contain the character `:`. Clients **MUST** consider a URI as relative if and only if it does not contain the character `:`.
+    * (**RECOMMENDED**) or `<name>` where `<name>` is defined as above. In this case, the Asset URL **MUST** ends with `#arc3`.
+* *Asset URL* (`au`): a URI pointing to a JSON Metadata file. 
+    * This URI as well as any URI in the JSON Metadata file:
+        * **SHOULD** be persistent and allow to download the JSON Metadata file forever.
+        * **MAY** contain the string `{id}`. If `{id}` exists in the URI, clients **MUST** replace this with the asset ID in decimal form. The rules below applies after such a replacement.
+        * **MUST** follow [RFC-3986](https://www.ietf.org/rfc/rfc3986.txt) and **MUST NOT** contain any whitespace character
+        * **SHOULD** use one of the following URI schemes (for compatibility and security): *https* and *ipfs*:
```

Comment:
I would like to see acknowledgement of schemes like ipns:// - this was would allow mutable metadata but still all be backed by IPFS.

Reply:
I just wouldn't want to define a metadata hash as it wouldn't be fixed.  The URL would point to an ipns link and the ARC3 metadata would be dereferenced via ipns [dnslink] -> ipfs CIDs.  This data would intentionally be mutable but being stored via ipfs would have decentralization benefits.
Personally, I think if ARC3 is going to draw a line in the sand that the metadata must be completely immutable I think that's an issue for some use cases.